core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -317,6 +327,15 @@ public void onComplete() {
         }
 
         if (oldState != State.DONE) {
+            if (http1WebSocketEstablished) {
+                if (isWritable()) {
+                    final ChannelFuture channelFuture = writeEmptyData();
+                    channelFuture.addListener(writeDataFutureListener(true, true));
+                    channelFuture.addListener(HttpServerHandler.CLOSE);
```

Comment:
Should we just call `disconnectWhenFinished()` to close the connection when the session is closed or considering to add the logic to https://github.com/line/armeria/blob/63791b04253c12fae463edcda7ca75eadd899c04/core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java#L435-L438